qolmat/audit/holes_characterization.py

```diff
@@ -0,0 +1,79 @@
+from __future__ import annotations
+from typing import Literal, Optional, TYPE_CHECKING
+
+import numpy as np
+import pandas as pd
+from scipy.stats import chi2
+
+from qolmat.imputations.imputers import ImputerEM
+
+if TYPE_CHECKING:
+    from qolmat.imputations.imputers import _Imputer
+
+
+class MCARTest:
+    """
+    This class implements the statistical tests to test the MCAR case.
+
+    Parameters
+        ----------
+        method : Literal[&quot;little&quot;]
+            The name of the statistical test. This should be handled by qolmat.
+        imputer : Optional[_Imputer], optional
+            If the selected test needs a imputer, you can provide the Imputer you want. Otherwise,
+            a default imputer will be used.
+    """
+
+    def __init__(self, method: Literal["little"], imputer: Optional[_Imputer] = None):
+        if method not in ["little"]:
+            raise ValueError(f"method` must be handled by qolmat, provided value is '{method}'")
+
+        self.method = method
+        self.imputer = imputer
+
+    def test(self, df: pd.DataFrame) -> float:
+        if self.method == "little":
+            return self.little_mcar_test(df)
```

Comment:
La méthode test ne regarde que "little" et dans le cas contraire ne fait rien. Que cela change-t-il par rapport à l'init. Puis si d'autres méthodes sont ajoutées ultérieurement, cette méthode doit être adaptée pour les gérer

Reply:
si demain par exemple, on ajoute PKLM. On devra ajouter cette méthode dans la fonction test.

L'idée c'est de ne préciser que à l'initialisation de la class MCAR quelle méthode on veut, puis ensuite quelque soit le méthode faire : 
mcar = MCAR("my_method")
mcar.test(my_df, **kwargs)

T'en penses quoi @Gsaes

Reply:
Je ne sais pas plus ce que je t'ai dit la dernière fois... mais le plus propre me semble de'être de passer par de l'héritage

class MCARTest:
    """
    This class implements the statistical tests to test the MCAR case.
    """

    def __init__(self):
        pass

    def test(self, df: pd.DataFrame) -> float:
        raise NotImplementedError("...")
        

class LittleTest(MCARTest):
    """docstring"""
    def __init__(self, imputer: Optional[_Imputer] = None):
        super().__init__()
        self.imputer = imputer

    def test(self, df: pd.DataFrame) -> float:
        """
        This method implements the Little's test. ...
        """
        < little_mcar_test>

Reply:
On peut ne pas laisser le choix et forcer l'utilisateur à utiliser l'ago EM. Est-ce que ça vaut le coup de lui laisser le choix de choisir ses paramètres ?